index.bs

```diff
@@ -1519,52 +2215,155 @@ partial interface MLGraphBuilder {
+    ::
+        An {{MLConvTranspose2dFilterOperandLayout}} [=string=].
+        Specifies the layout format of the filter tensor as follow:
+            - **"iohw"**: [input_channels, output_channels/groups, height, width]
+            - **"hwoi"**: [height, width, output_channels/groups, input_channels]
+            - **"ohwi"**: [output_channels/groups, height, width, input_channels]
+        The default value is *"iohw"*.
+
+    : <dfn>bias</dfn>
+    ::
+        An {{MLOperand}} object.
+        Specifies the additional 1-D tensor with the shape of *[output_channels]* whose values are to be added to the convolution result.
 
-    *output size = (input size - 1) ** *stride + (filter size - 1) ** *dilation + 1 - beginning padding - ending padding + output padding*
+    : <dfn>activation</dfn>
+    ::
+        An {{MLActivation}} object.
+        Specifies the optional activation function that immediately follows the convolution operation.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input 4-D tensor. The logical shape
+            is interpreted according to the value of *options*.{{MLConvTranspose2dOptions/inputLayout}}.
+        - *filter*: an {{MLOperand}}. The filter 4-D tensor. The logical shape is
+            interpreted according to the value of *options*.{{MLConvTranspose2dOptions/filterLayout}} and {{MLConvTranspose2dOptions/groups}}.
+        - *options*: an optional {{MLConvTranspose2dOptions}}.
+
+    **Returns:** an {{MLOperand}}. The output 4-D tensor that contains the transposed convolution result. The output shape is interpreted according to the *options*.{{MLConvTranspose2dOptions/inputLayout}} value. More specifically, unless the *options*.{{MLConvTranspose2dOptions/outputSizes}} values are explicitly specified, the *options*.{{MLConvTranspose2dOptions/outputPadding}} may be needed to compute the spatial dimension values of the output tensor as follow:
+
+    *output_size = (input_size - 1) ** *stride + (filter_size - 1) ** *dilation + 1 - beginning_padding - ending_padding + output_padding*
 </div>
 
+<details open>
+  <summary>
+    The {{MLGraphBuilder/convTranspose2d(input, filter, options)}} steps are:
+  </summary>
+  <div algorithm=convtranspose2d class=algorithm-steps>
+    1. If |input| or |filter| is not an instance of {{MLOperand}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |input_size| be the size of |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+    1. Let |filter_size| be the size of |filter|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+    1. If |input_size| is not `4`, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. If |filter_size| is not `4`, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. If |options| is `undefined`, let |options| be an empty [=object=].
+    1. If |options|.{{MLConvTranspose2dOptions/padding}} is `undefined`, set it to `[0, 0, 0, 0]`.
```

Comment:
Yes, and also other checks that now have been added to conv2d.

Note to self. I will also investigate if we can factor out checking these options into separate validation algorithms. Actually I am sure it can be done, but its higher priority to merge this PR ASAP when it's good enough (though not yet optimal).